src/sage/algebras/lie_algebras/representation.py

```diff
@@ -791,3 +791,242 @@ def _acted_upon_(self, scalar, self_on_left=False):
                 return P._project(P._to_pbw(scalar) * self._lift_pbw())
 
             return super()._acted_upon_(scalar, self_on_left)
+
+
+class FaithfulRepresentationPBWPosChar(CombinatorialFreeModule, Representation_abstract):
+    r"""
+    A faithful representation of a finite dimensional Lie algebra
+    in positive characteristic.
+
+    .. WARNING::
+
+        This is often a very large dimensional representation relative
+        to the dimension of the Lie algebra.
+
+    ALGORITHM:
+
+    We implement the algorithm given in [deG2000] Section 6.6. Let `L`
+    be a finite dimensional Lie algebra over a ring of characteristic `p`
+    with basis `(b_1, \ldots, b_n)`. We compute (monic) `p`-polynomials
+    `f_i` such that `A = \mathrm{ad}(b_i)` solves `f_i(A) = 0` by using
```

Comment:
Need to explain what "ad" is?

Reply:
I explained a bit but I just used the full terminology. I expect anyone reading this should know what the adjoint action is, or at least can very quickly find it.

Reply:
Thanks, that's good enough